articles/active-directory/develop/quickstarts/active-directory-windesktop.md

```diff
@@ -0,0 +1,142 @@
+> ## Register your application and download your quickstart app
+> You have two options to start your quickstart application:
+> * [Express] [Option 1: Register and auto configure your app and then download your code sample](#option-1-register-and-auto-configure-your-app-and-then-download-your-code-sample)
+> * [Manual] [Option 2: Register and manually configure your application and code sample](#option-2-register-and-manually-configure-your-application-and-code-sample)
+>
+> ### Option 1: Register and auto configure your app and then download your code sample
+>
+> 1. Go to the [Azure portal - Application Registration](https://portal.azure.com/signin/index/?Microsoft_AAD_RegisteredApps=true#blade/Microsoft_AAD_RegisteredApps/applicationsListBlade/quickStartType/WinDesktopQuickstartPage/sourceType/docs).
+> 1. Enter a name for your application and click **Register**.
+> 1. Follow the instructions to download and automatically configure your new application for you in one click.
+>
+> ### Option 2: Register and manually configure your application and code sample
+>
+> #### Step 1: Register your application
+> 
+> 1. To register an application for this code sample, go to the [Azure portal - Application Registration](https://aka.ms/registeredappsprod) and select **New registration**.
+> 1. Enter a name for your application, add `urn:ietf:wg:oauth:2.0:oob` in **Reply URL**, and click **Register**.
+
+> [!div renderon="portal" class="sxs-lookup"]
+> #### Step 1: Configure your application
+> For the code sample for this quickstart to work, you need to add a reply URL as **urn:ietf:wg:oauth:2.0:oob**.
+> > [!div renderon="portal" id="makechanges" class="nextstepaction"]
+> > [Make this change for me]()
+>
+> > [!div id="appconfigured" class="hidden"]
+> > ![Already configured](../../../../includes/media/active-directory-develop-common/green-check.png) Your application is configured with these attributes
+
+#### Step 2: Download your Visual Studio project
+
+ [Download the Visual Studio 2017 project](https://github.com/Azure-Samples/active-directory-dotnet-desktop-msgraph-v2/archive/master.zip)
+
+#### Step 3: Configure your Visual Studio project
+
+ 1. Open the project in Visual Studio
+ 1. Edit **App.Xaml.cs** and replace the line starting with `private static string ClientId` with the Application ID from the application you just registered:
+
+    ```csharp
+    private static string ClientId = "Enter_the_Application_Id_here";
+    ```
+
+## More Information
+
+### MSAL.NET
+
+MSAL ([Microsoft.Identity.Client](https://www.nuget.org/packages/Microsoft.Identity.Client)) is the library used to sign in users and request tokens used to access an API protected by Microsoft Azure Active Directory. You can install it by running the following command in Visual Studio's *Package Manager Console*:
+
+```powershell
+Install-Package Microsoft.Identity.Client -Pre
+```
+
+### MSAL Initialization
+
+You can add the reference for MSAL by adding the line below:
```

Comment:
This is not adding a reference. It's just adding a namespace to avoid developers to have to use the fully qualified name. I would rather write "you can reference MSAL's namespace by adding the line below"